src/garage/_dtypes.py

```diff
@@ -167,3 +166,128 @@ def __new__(cls, env_spec, observations, actions, rewards, terminals,
         return super().__new__(TrajectoryBatch, env_spec, observations,
                                actions, rewards, terminals, env_infos,
                                agent_infos, lengths)
+
+
+class Sample(
+        collections.namedtuple('Sample', [
+            'env_spec',
+            'observation',
+            'action',
+            'reward',
+            'next_observation',
+            'terminal',
+            'env_info',
+            'agent_info',
+        ])):
+    # pylint: disable=missing-return-doc, missing-return-type-doc, missing-param-doc, missing-type-doc  # noqa: E501
+    r"""A tuple representing a single sample.
+
+    A :class:`Sample` represents a single sample when an agent interacts with
+        an environment.
+
+    Attributes:
+        env_spec (garage.envs.EnvSpec): Specification for the environment from
+            which this data was sampled.
+        observation (numpy.ndarray): A numpy array of shape
+            :math:`(O^*)` containing the observation for the current time step
```

Comment:
can you ensure these are consistently wrapped to the 78-char line?

Reply:
These are all wrapped to 79 chars.

Reply:
i was referring to indenting things like `:math:`(O^*)` too early, creating lots of whitespace on the right.